Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -231,6 +231,7 @@ void WiiSocket::Update(bool read, bool write, bool except)
         WiiSockMan::Convert(*wii_name, local_name);
 
         int ret = connect(fd, (sockaddr*)&local_name, sizeof(local_name));
+        ret = WiiSockMan::GetInstance().AddSocket(ret);
```

Comment:
`connect` return 0 on success or -1 on error; it doesn't return a socket. Did you want to call this on `fd` instead of `ret`?